pkg/kubectl/cmd/drain.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+type DrainOptions struct {
+	NodeName           string
+	Force              bool
+	GracePeriodSeconds int
+}
+
+const (
+	drain_long = `Drain node in preparation for maintenance.
+
+The given node will be marked unschedulable to prevent new pods from arriving.
+Then all running pods will be removed from the machine.  If there are any
+unreplicated pods, no pods will be touched without --force.
+
+When you are ready to put the node back into service, use kubectl unlame.`
+	drain_example = `# Drain node "foo", even if there are unreplicated pods on it.
+$ kubectl drain foo --force
+
+# As above, but don't delete unreplicated pods, and use a grace period of 15 minutes.
+$ kubectl drain foo --grace-period=900
+`
+)
+
+func NewCmdDrain(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := &DrainOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "drain NODE",
```

Comment:
I would suggest calling this drainmachine or something like that, so that you don't have to worry about backward compatibility when you implement the more sophisticated features you plan to implement in the future (e.g. automatically iterate through the machines, respect SLOs, move the logic to the server, etc.) -- you could call that command drain, and still keep drainmachine around.